lib/puppet/provider/package/aix.rb

```diff
@@ -106,8 +119,12 @@ def self.pkglist(hash = {})
     end
 
     begin
-      list = execute(cmd).scan(/^[^#][^:]*:([^:]*):([^:]*)/).collect { |n,e|
-        { :name => n, :ensure => e, :provider => self.name }
+      list = execute(cmd).scan(/^[^#][^:]*:([^:]*):([^:]*):[^:]*:[^:]*:([^:])/).collect { |n,e,s|
+        if [:broken, :inconsistent].include?(STATE_CODE[s])
+          Puppet.warning("Package #{n} is in a #{STATE_CODE[s]} state and requires manual intervention.")
```

Comment:
Won't this generate too much noise even if the users changed nothing?

Reply:
I thought about this, but on a fresh AIX box there weren't any broken packages (out of hundreds), so it might be useful to report them somehow if present. I also thought about lowering the severity to debug, but then the logs would easily get mixed up with others.

This will still exit with 0, but show differently in PE I think. I'll check if the logs don't show up multiple times if managing multiple resources.

Reply:
So, looks like these warnings aren't even shown when applying manifests like:

```puppet
package {'openssl.base':
  ensure => 'present',
  source => '/root/openssl-1.0.2.1800',
}

package {'openssl.license':
  ensure => 'present',
  source => '/root/openssl-1.0.2.1800',
}
```
```
Debug: Executing: '/usr/bin/lslpp -qLc openssl.base'
Debug: Executing: '/usr/bin/lslpp -qLc openssl.man.en_US'
Debug: Executing: '/usr/bin/lslpp -qLc openssl.license'
```

Since puppet executes lslpp with the package name, the broken packages won't be shown. The warnings will only be shown when running `puppet resource package`.

Reply:
I'll remove the warning though... since it will also error when ensuring a broken package:

```sh-session
> puppet resource package bos.adt.debug --param provider ensure=present source=/tmp/a   
Warning: Package 'bos.adt.debug' is in a inconsistent state and requires manual intervention
Error: Package 'bos.adt.debug' is in a inconsistent state and requires manual intervention
Error: /Package[bos.adt.debug]/ensure: change from 'absent' to 'present' failed: Package 'bos.adt.debug' is in a inconsistent state and requires manual intervention
package { 'bos.adt.debug':
  ensure   => 'absent',
  provider => 'aix',
}
```